File: 03-strings/08_longest_common_prefix.py
import sys, os; sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
from test_runner import run_tests


# Longest Common Prefix (LC #14) -- Easy
# Find longest common prefix among array of strings.
#
#   Key: Vertical scanning -- compare char by char across all strings

def longest_common_prefix(strs):
    pass


run_tests(longest_common_prefix, [
    ((["flower","flow","flight"],), "fl"),
    ((["dog","racecar","car"],),    ""),
    ((["a"],),                      "a"),
    ((["","b"],),                   ""),
])

File: 03-strings/08_longest_common_prefix.c
#include "../test_runner.h"

void longest_common_prefix(char** strs, int n, char* out) {
    out[0] = '\0';
}

int main() {
    char out[256];
    char* s1[] = {"flower","flow","flight"};
    longest_common_prefix(s1, 3, out); TEST_STR("fl", out, "fl");
    char* s2[] = {"dog","racecar","car"};
    longest_common_prefix(s2, 3, out); TEST_STR("none", out, "");
    TEST_SUMMARY();
}
